components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/ClaimMetadataManagementServiceImpl.java

```diff
@@ -229,6 +230,31 @@ public void updateLocalClaim(LocalClaim localClaim, String tenantDomain) throws
         ClaimMetadataEventPublisherProxy.getInstance().publishPostUpdateLocalClaim(tenantId, localClaim);
     }
 
+    /**
+     * Update attribute mappings for local claims in bulk
+     *
+     * @param localClaimList
+     * @param tenantDomain
+     * @param userStoreDomain
+     * @throws ClaimMetadataException
+     */
+    public void updateLocalClaimMappings(List<LocalClaim> localClaimList, String tenantDomain, String userStoreDomain)
+            throws ClaimMetadataException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        ClaimMetadataEventPublisherProxy claimMetadataEventPublisherProxy =
+                ClaimMetadataEventPublisherProxy.getInstance();
+        for (LocalClaim localClaim : localClaimList) {
+            claimMetadataEventPublisherProxy.publishPreUpdateLocalClaim(tenantId, localClaim);
+        }
+
+        this.localClaimDAO.updateLocalClaimMappings(localClaimList, tenantId, userStoreDomain);
+
+        for (LocalClaim localClaim : localClaimList) {
+            ClaimMetadataEventPublisherProxy.getInstance().publishPostUpdateLocalClaim(tenantId, localClaim);
```

Comment:
We can use the same local variable here as well.

Reply:
Addressed with commit abcc28ca2bde2f2eea597841ec5000f0cddd7202